x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/job/process/AbstractProcessWorkerExecutorService.java

```diff
@@ -112,7 +111,6 @@ public void start() {
                     } catch (Exception e) {
                         logger.error(() -> "error handling process [" + processName + "] operation", e);
                     }
-                    EsExecutors.rethrowErrors(ThreadContext.unwrap(runnable));
```

Comment:
@davidkyle should I remove references to `preserveContext` too?

Reply:
After chatting with Dave R it seems like we should keep the preserved context.

Reply:
Yes the code always `unwrap()`. The thread context is the mechanism for preserving headers so we get things like deprecated warnings in the REST API.

https://github.com/elastic/elasticsearch/blob/main/x-pack/plugin/src/yamlRestTest/resources/rest-api-spec/test/ml/3rd_party_deployment.yml#L249


However, I think the `EsExecutors.rethrowErrors(` part is redundant and confusing as it only applies to RunnableFutures which are not used here and the InterruptedException is handled in this class. How about adding an assertion that the Runnable is not an instance of RunnableFuture then the `rethrowErrors` can safely be removed